prow/plank/reconciler.go

```diff
@@ -944,3 +975,26 @@ func isRequestError(err error) bool {
 	}
 	return 400 <= code && code < 500
 }
+
+func countPendingOrOlderMatchingPJs(pj prowv1.ProwJob, pjs []prowv1.ProwJob) int {
+	var pendingOrOlderMatchingPJs int
+
+	for _, foundPJ := range pjs {
+		// Ignore self here.
+		if foundPJ.UID == pj.UID {
+			continue
+		}
+		if foundPJ.Status.State == prowv1.PendingState {
+			pendingOrOlderMatchingPJs++
+			continue
+		}
+
+		// At this point we know that foundPJ is in Triggered state, if its older than our prowJobs it gets
```

Comment:
Why do we know this about the state? Why is it not errored? Why not check explicitly?

Reply:
Good catch. This is my bad. This function used to be called in context of using `optPendingTriggeredJobsNamed`. That is why there is no explicit check for the `Triggered` state.

I added the check and renamed function to `countPendingOrOlderTriggeredMatchingPJs`.